src/include/network/services/routing.rb

```diff
@@ -424,43 +438,65 @@ def validateRouting(_key, _event)
       false
     end
 
+    # Mangles route definition for storing in routes file
+    #
+    # Basically netmask field is obsolete, so it converts
+    # the record to use CIDR notation if netmask is defined.
+    #
+    # For details see "man routes".
+    def convert_route_conf(route)
+      return route if route["netmask"].empty? || route["netmask"] == "-"
+
+      dest = route["destination"]
+      netmask = route["netmask"]
+
+      if Netmask.Check4(netmask)
+        cidr = Netmask.ToBits(netmask)
+      elsif netmask.start_with?("/")
+        cidr = netmask[1..-1]
+      else
+        # if it is netmask then long netmask is not supported for IPv6
+        # if it is prefix length (CIDR), then prefix it has to be prefixed by '/'
+        raise ArgumentError, "Invalid netmask or prefix length"
+      end
+
+      route["destination"] = "#{dest}/#{cidr}"
+      route["netmask"] = "-"
+
+      route
+    end
+
     def storeRouting(_key, _event)
-      route_conf = Builtins.maplist(@r_items) do |e|
-        {
-          "destination" => Ops.get_string(e, 1, ""),
-          "gateway"     => Ops.get_string(e, 2, ""),
-          "netmask"     => Ops.get_string(e, 3, ""),
-          "device"      => Ops.get_string(e, 4, ""),
-          "extrapara"   => Ops.get_string(e, 5, "")
-        }
+      route_conf = @r_items.map do |route|
+        convert_route_conf(
+          "destination" => route[1].to_s,
+          "gateway"     => route[2].to_s,
+          "netmask"     => route[3].to_s,
+          "device"      => route[4].to_s,
+          "extrapara"   => route[5].to_s
+        )
       end
 
       @defgw = UI.QueryWidget(Id(:gw), :Value)
       @defgwdev = UI.QueryWidget(Id(:gw4dev), :Value)
       @defgw6 = UI.QueryWidget(Id(:gw6), :Value)
       @defgwdev6 = UI.QueryWidget(Id(:gw6dev), :Value)
 
-      if @defgw != ""
-        route_conf = Builtins.add(
-          route_conf,
-          "destination" => "default",
-          "gateway"     => @defgw,
-          "netmask"     => "-",
-          "device"      => @defgwdev
-        )
-      end
-
-      if @defgw6 != ""
-        route_conf = Builtins.add(
-          route_conf,
-          "destination" => "default",
-          "gateway"     => @defgw6,
-          "netmask"     => "-",
-          "device"      => @defgwdev6
-        )
-      end
-
-      Routing.Routes = deep_copy(route_conf)
+      route_conf << {
+        "destination" => "default",
+        "gateway"     => @defgw,
+        "netmask"     => "-",
+        "device"      => @defgwdev
+      } if !@defgw.empty?
+
+      route_conf << {
+        "destination" => "default",
+        "gateway"     => @defgw6,
+        "netmask"     => "-",
+        "device"      => @defgwdev6
+      } if !@defgw6.empty?
```

Comment:
just question, why @defgw6 and others are instance variables and not local ones?